### ISSUE TITLE
Syncthing-Android for iOS via Flutter?

### ISSUE THREAD

**Issue by oneyb** [syncthing/syncthing-android#1229]
Dear @Nutomic & Co.,

I know my issue title is bs. I am picking up Dart with the hope of becoming proficient in the Flutter framework. 

I wonder if you have discussed the possibility of maintaining a single codebase which would compile packages for both iOS and android (and in the future for fuchsia)?

Have you considered flutter for just for syncthing-android?

Cheers
Brian

**Comment by Nutomic** [comment]
Is there any way to compile Syncthing itself for iOS? And even if that works, it would take a lot of work to port all the Java code here to Dart. 

Anyway, I have no idea about iOS or anything Apple, so I'm probably the wrong guy to ask. It might be better if you make a thread on the forum about this.

**Comment by calmh** [comment]
Yes. Syncthing can run on iOS, but it cannot access files from other apps. That means, among other things, accessing photos and media, password managers, things like that. So usefulness is limited. It really needs more of a rewrite to make sense.

**Comment by Nutomic** [comment]
In that case, it might be possible to port [syncthing-java](https://github.com/syncthing/syncthing-java) to [Kotlin Native](https://kotlinlang.org/docs/reference/native-overview.html). It already implements the protocol from scratch, and lets you upload/download individual files. 

Using syncthing-java has the advantage that it's explicitly written for mobile devices, and doesn't implement continuous sync (which is hard on mobile). And you could write the frontend in any language you like.
